worker/migrationmaster/worker.go

```diff
@@ -582,7 +583,10 @@ func (w *Worker) transferLogs(targetInfo coremigration.TargetInfo, modelUUID str
 
 	throwWrench := latestLogTime == utcZero && wrench.IsActive("migrationmaster", "die-after-500-log-messages")
 
-	logSource, err := w.config.Facade.StreamModelLog(latestLogTime)
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	logSource, err := w.config.Facade.StreamModelLog(ctx, latestLogTime)
```

Comment:
So the net sum of all these changes is just this, right? That if the model migration worker exits before finishing streaming, it will kill the associated goroutine.
Is there a reason to do this around contexts rather than Tombs (which predated context, which is why it is everywhere). I don't have a problem with context, just wondering if we would already have tomb objects around that we weren't properly utilizing.

Reply:
I find they serve slightly different purposes. I tend to think that `context.Context` gives a better API for passing them, where tombs feel like they're internal implementations.